pennylane/ops/qubit/matrix_ops.py

```diff
@@ -142,16 +154,24 @@ def compute_decomposition(U, wires):
         """
         # Decomposes arbitrary single-qubit unitaries as Rot gates (RZ - RY - RZ format),
         # or a single RZ for diagonal matrices.
-        if qml.math.shape(U) == (2, 2):
+        shape = qml.math.shape(U)
+        if shape == (2, 2):
             return qml.transforms.decompositions.zyz_decomposition(U, Wires(wires)[0])
 
-        if qml.math.shape(U) == (4, 4):
+        if shape == (4, 4):
             return qml.transforms.two_qubit_decomposition(U, Wires(wires))
 
+        # TODO[dwierichs]: Implement decomposition of broadcasted unitary
+        if len(shape) == 3:
+            raise DecompositionUndefinedError(
+                "The decomposition of QubitUnitary does not support broadcasting."
+            )
```

Comment:
What would need to happen here to support this? I suppose, at this point, multiple tapes must be created?

Reply:
> All of this is not crazy-complicated. Should I include it in this PR?

I would create an issue, and leave it for another PR :)